pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -0,0 +1,112 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import torch
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins.base_plugin import Plugin
+
+
+class TrainingTypePlugin(Plugin, ABC):
+    """A Plugin to change the behaviour of the training, validation and test-loop."""
```

Comment:
@justusschock - another question: currently [these training loop functions](https://github.com/PyTorchLightning/pytorch-lightning/blob/8b4fd44563b72c3351e0cc5c4a2c6a848d123cd4/pytorch_lightning/trainer/trainer.py#L496-L502) are set on the accelerator backend in `fit()` - should these be represented in the interface here or on the accelerator?  
What do you think about the training type plugin accepting these as constructor arguments, assuming we can define some base interfaces for train/evaluation/test loop?

Reply:
I don't think this is needed anymore, since we pass the Trainer to start_training etc, where we can directly call trainer.train